src/evalrule.jl

```diff
@@ -1,73 +1,43 @@
-###########################################################################
+# integration segment (a,b), estimated integral I, and estimated error E
+struct Segment{TX,TI,TE}
+    a::TX
+    b::TX
+    I::TI
+    E::TE
+end
+Base.promote_rule(::Type{Segment{TX,TI,TE}}, ::Type{Segment{TX′,TI′,TE′}}) where {TX,TI,TE,TX′,TI′,TE′} = Segment{promote_type(TX,TX′), promote_type(TI,TI′), promote_type(TE,TE′)}
```

Comment:
```suggestion
Base.promote_rule(::Type{Segment{TX,TI,TE}}, ::Type{Segment{TX′,TI′,TE′}}) where {TX,TI,TE,TX′,TI′,TE′} =
    Segment{promote_type(TX,TX′), promote_type(TI,TI′), promote_type(TE,TE′)}
```
Just a long line nitpick, feel free to ignore if you prefer.

Reply:
ok